lib/color.rb

```diff
@@ -1,7 +1,17 @@
 # Colorization for strings
 class String
+  @use_color = STDOUT.isatty
+
+  class << self
+    attr_accessor :use_color
+  end
+
   def colorize(color_code, shade)
-    "\e[#{shade};#{color_code}m#{self}\e[0m"
+    if self.class.use_color
+      return "\e[#{shade};#{color_code}m#{self}\e[0m"
+    else
+      return self
```

Comment:
Is this necessary?  What would happen if it was ignored?

Reply:
I think so. This is the heart of the functionality. The on/off switch for colorization is a class instance variable, so it _can't_ be ignored by the colorization functions. If someone explicitly outputs a colorizing escape sequence in crew or a package.rb, I'd call that a defect. If build output for a package outputs color, I'm not sure that there's much we can do to prevent that...